### PR TITLE
[Flax] Correct flax docs

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Lots of Flax models had incorrect or missing documentation. In this PR I went through the docs of all Flax models and corrected the docstrings so that they:

- run correctly
- show the correct use case (some MBart examples didn't make sense)
- link to the correct output types

@LysandreJik @sgugger - it would be great if we could re-enable the examples tests to make sure they all run correctly. Happy to help on it this week!

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: docs/source/model_doc/wav2vec2.rst [hunk 1]
```diff
@@ -67,6 +67,22 @@ Wav2Vec2Processor
...
+
+.. autoclass:: transformers.models.wav2vec2.modeling_wav2vec2.Wav2Vec2ForPreTrainingOutput
+    :members: 
+
+.. autoclass:: transformers.models.wav2vec2.modeling_flax_wav2vec2.FlaxWav2Vec2BaseModelOutput
+    :members: 
+
+.. autoclass:: transformers.models.wav2vec2.modeling_flax_wav2vec2.FlaxWav2Vec2ForPreTrainingOutput
+    :members: 
+
```
Comment (sgugger) [inline docs/source/model_doc/wav2vec2.rst:84]: There are TF classes as well no?

**Review by sgugger — APPROVED** [review]
Thanks for the cleanup! Doc tests are definitely on the roadmap for very soon.

File: src/transformers/file_utils.py [hunk 2]
```diff
@@ -1115,9 +1115,10 @@ def _prepare_output_docstrings(output_type, config_class):
         >>> model = {model_class}.from_pretrained('{checkpoint}')
 
         >>> inputs = tokenizer("Hello, my dog is cute", return_tensors="jax")
```
Comment (patil-suraj) [inline src/transformers/file_utils.py:1117]: (nit)
Think we should return `np` tensors instead of `jax`, as we have seen during the sprint that if pre-processing functions which use JAX are called in multi-processing code such as our `dataset.map` it creates issues. It's not an issue here but would be nice to if always encourage to use `numpy` when pre-processing.

File: docs/source/model_doc/wav2vec2.rst (hunk 1 above)
Comment (patrickvonplaten) [reply docs/source/model_doc/wav2vec2.rst:84]: Haven't implemented the `TFWav2Vec2ForPreTraining` class yet as it's quite complicated (gumbel softmax, contrastive loss, ...)

File: src/transformers/models/t5/modeling_flax_t5.py [hunk 3]
```diff
@@ -844,6 +851,72 @@ def __call__(
...
+            Contains precomputed key and value hidden states of the attention blocks. Can be used to speed up decoding.
+
+            If :obj:`past_key_values` are used, the user can optionally input only the last :obj:`decoder_input_ids`
+            (those that don't have their past key value states given to this model) of shape :obj:`(batch_size, 1)`
+            instead of all :obj:`decoder_input_ids` of shape :obj:`(batch_size, sequence_length)`.
+
+        use_cache (:obj:`bool`, `optional`):
+            If set to :obj:`True`, :obj:`past_key_values` key value states are returned and can be used to speed up
+            decoding (see :obj:`past_key_values`).
+
```
Comment (patil-suraj) [inline src/transformers/models/t5/modeling_flax_t5.py:908]: we don't use `use_cache`, no? since we always cache the past key/values.

**Review by patil-suraj — APPROVED** [review]
Thanks a lot for fixing this!

File: docs/source/model_doc/wav2vec2.rst (hunk 1 above)
Comment (sgugger) [reply docs/source/model_doc/wav2vec2.rst:84]: Yes but there is the base model, no?

File: src/transformers/file_utils.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/transformers/file_utils.py:1117]: Agree! Think this should be a follow-up PR to correct this everywhere (examples, notebooks, blog posts, ...)

File: src/transformers/models/t5/modeling_flax_t5.py (hunk 3 above)
Comment (patrickvonplaten) [reply src/transformers/models/t5/modeling_flax_t5.py:908]: good catch!

File: src/transformers/models/t5/modeling_flax_t5.py [hunk 4]
```diff
@@ -844,6 +851,72 @@ def __call__(
...
+            sequence of hidden states at the output of the last layer of the encoder. Used in the cross-attention of
+            the decoder.
+        past_key_values (:obj:`tuple(tuple(jnp.ndarray))` of length :obj:`config.n_layers` with each tuple having 4 tensors of shape :obj:`(batch_size, num_heads, sequence_length - 1, embed_size_per_head)`):
+            Contains precomputed key and value hidden states of the attention blocks. Can be used to speed up decoding.
+
+            If :obj:`past_key_values` are used, the user can optionally input only the last :obj:`decoder_input_ids`
+            (those that don't have their past key value states given to this model) of shape :obj:`(batch_size, 1)`
+            instead of all :obj:`decoder_input_ids` of shape :obj:`(batch_size, sequence_length)`.
+
+        use_cache (:obj:`bool`, `optional`):
```
Comment (patrickvonplaten) [inline src/transformers/models/t5/modeling_flax_t5.py:905]:
```suggestion
```

File: src/transformers/models/t5/modeling_flax_t5.py [hunk 5]
```diff
@@ -844,6 +851,72 @@ def __call__(
...
+            If set to :obj:`True`, :obj:`past_key_values` key value states are returned and can be used to speed up
```
Comment (patrickvonplaten) [inline src/transformers/models/t5/modeling_flax_t5.py:906]:
```suggestion
```

File: src/transformers/models/t5/modeling_flax_t5.py [hunk 6]
```diff
@@ -844,6 +851,72 @@ def __call__(
...
+            decoding (see :obj:`past_key_values`).
```
Comment (patrickvonplaten) [inline src/transformers/models/t5/modeling_flax_t5.py:907]:
```suggestion
```
